### PR TITLE
[201911][DellEMC] Fix abrupt reboot in S6000

### PR DESCRIPTION
**Pull request by rkdevi27**
#### Why I did it
The S6000 devices, the cold reboot is abrupt and it is likely to cause issues which will cause the device to land into EFI shell. Hence the platform reboot will happen after graceful unmount of all the filesystems as in S6100.
#### How I did it
Moved the platform_reboot to platform_reboot_override and hooked it to the systemd shutdown services as in S6100
#### How to verify it
Issue "reboot" command to verify if the reboot is happening gracefully.
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog

Dell S6000 graceful reboot
[UT_logs_201911.txt](https://github.com/Azure/sonic-buildimage/files/6049162/UT_logs_201911.txt)

Please cherry pick the changes from https://github.com/Azure/sonic-utilities/pull/1454 into 201911 branch

### CODE REVIEW

**Review by jleveque — APPROVED** [review]

**Comment by jleveque** [comment]
Retest this please

File: platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import struct
+
+PORT_RES = '/dev/port'
+COLD_RESET = 0xE     # Cold Reset 
+WARM_RESET = 0x6     # Warm Reset 
+RESET_REG = 0xCF9
+
+def io_reg_write(resource, offset, val):
+    fd = os.open(resource, os.O_RDWR)
+    if(fd < 0):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override:13]: Parentheses unnecessary. Please remove.

File: platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        print 'file open failed %s" % resource'
+        return
+    if(os.lseek(fd, offset, os.SEEK_SET) != offset):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override:16]: Parentheses unnecessary. Please remove.

File: platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        print 'lseek failed on %s' % resource
+        return
+    ret = os.write(fd, struct.pack('B', val))
+    if(ret != 1):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override:20]: Parentheses unnecessary. Please remove.

File: platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_update_reboot_cause [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import os
+import struct
+
+NVRAM_RES = '/dev/nvram'
+COLD_RESET = 0xE     # Cold Reset 
+WARM_RESET = 0x6     # Warm Reset 
+
+def io_reg_write(resource, offset, val):
+    fd = os.open(resource, os.O_RDWR)
+    if(fd < 0):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_update_reboot_cause:12]: Parentheses unnecessary. Please remove.

File: platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_update_reboot_cause [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        print 'file open failed %s" % resource'
+        return
+    if(os.lseek(fd, offset, os.SEEK_SET) != offset):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_update_reboot_cause:15]: Parentheses unnecessary. Please remove.

File: platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_update_reboot_cause [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        print 'lseek failed on %s' % resource
+        return
+    ret = os.write(fd, struct.pack('B', val))
+    if(ret != 1):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_update_reboot_cause:19]: Parentheses unnecessary. Please remove.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override (hunk 1 above)
Comment (rkdevi27) [reply platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override:13]: Thank You. I have used the existing platform_reboot script (which has been removed as part of this PR) and edit it accordingly. Will do the changes and update the PR.

Comment (sujinmkang) [reply platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override:13]: @rkdevi27 are you going to update this PR?

Comment (rkdevi27) [reply platform/broadcom/sonic-platform-modules-dell/s6000/scripts/platform_reboot_override]: > @rkdevi27 are you going to update this PR?

@sujinmkang  Updated the PR. Thanks.

**Comment by rlhui** [comment]
are there PRs that are the the equivalent of this for 201811, 201212 or master branch? Thanks.

**Comment by rkdevi27** [comment]
> are there PRs that are the the equivalent of this for 201811, 201212 or master branch? Thanks.



> are there PRs that are the the equivalent of this for 201811, 201212 or master branch? Thanks.

Yes, the PRs are already merged for 201811 and master branch.
https://github.com/Azure/sonic-buildimage/pull/6923 - PR that got merged in 201811
https://github.com/Azure/sonic-buildimage/pull/6835 - PR that got merged in master

**Review by jleveque — APPROVED** [review]

**Comment by jleveque** [comment]
@abdosi, @rlhui: Please merge at a convenient time.

**Pull request merged**